hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -452,6 +458,51 @@ protected HoodieRecord<T> updateFileName(HoodieRecord<T> record, Schema schema,
     return record.prependMetaFields(schema, targetSchema, metadataValues, prop);
   }
 
+  private void trackMetadataIndexStats(Option<HoodieKey> hoodieKeyOpt, Option<HoodieRecord> combinedRecordOpt, Option<HoodieRecord<T>> oldRecordOpt, boolean isDelete) {
+    if (!config.isSecondaryIndexEnabled() || secondaryIndexDefns.isEmpty() || !config.isMetadataStreamingWritesEnabled(hoodieTable.getMetaClient().getTableConfig().getTableVersion())) {
```

Comment:
based on earlier feedback, we should be able to access a protected method from HoodieWriteHandle for this

Reply:
Addressed